src/Concordium/Client/Types/Contract/Schema.hs

```diff
@@ -716,8 +790,21 @@ putLenWithSizeLen sl len = case sl of
 
 -- * Map *
 
-getMapOfWithSizeLen :: Ord k => SizeLength -> S.Get k -> S.Get v -> S.Get (Map k v)
-getMapOfWithSizeLen sl gt gv = S.label "Map" $ Map.fromList <$> getListOfWithSizeLen sl (S.getTwoOf gt gv)
+allUnique :: (Ord a) => [a] -> Bool
+allUnique xs = length (nub xs) /= length xs
+
+-- |Get a map with a specied size length. Fails if each element in the list of
+-- deserialized map tuples mapped over f is not unique.
+getMapOfWithSizeLenAndPred :: (Ord k) => ([(k,v)] -> Bool) -> SizeLength -> S.Get k -> S.Get v -> S.Get (Map k v)
```

Comment:
This mentions uniqueness, but the predicate could be used for checking other properties.
I would suggest describing that it takes a predicate for checking some property and that this fx could be uniqueness.
Please also update the error message a few lines below with the same idea in mind.